fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStoreBase.java

```diff
@@ -169,11 +170,41 @@ default FDBStoreTimer getTimer() {
     interface UserVersionChecker {
         /**
          * Check the user version.
+         * <p>
+         * If this store is being created for the first time, the store header will be set to the default instance. In
+         * particular, the format version will be zero, whereas the format version on all stores that have already been
+         * created will be greater than zero.
+         *
+         * @param storeHeader store header
+         * @param metaData the meta-data provider that will be used to get meta-data
+         *
+         * @return the user version to store in the record info header
+         */
+        default CompletableFuture<Integer> checkUserVersion(@Nonnull RecordMetaDataProto.DataStoreInfo storeHeader,
```

Comment:
```suggestion
        @SuppressWarnings("deprecation")
        default CompletableFuture<Integer> checkUserVersion(@Nonnull RecordMetaDataProto.DataStoreInfo storeHeader,
```

I believe this is needed because it now calls the deprecated `checkUserVersion`. (I also think our PRBs are currently configured _not_ to check this, which is a bit of a problem.)

Reply:
Added, though that resulted in "redundant suppression" warning in my Intellij (and there was no deprecated call warning in the first place). Maybe something is up with my configuration.